src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2276,124 +2276,139 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
                 // indexers, and non-conversion operators -- starts with a type 
                 // (possibly void). Parse that.
                 TypeSyntax type = ParseReturnType();
-                var sawRef = type.Kind == SyntaxKind.RefType;
 
-                // Check for misplaced modifiers.  if we see any, then consider this member
-                // terminated and restart parsing.
-                if (GetModifier(this.CurrentToken) != DeclarationModifiers.None &&
-                    this.CurrentToken.ContextualKind != SyntaxKind.PartialKeyword &&
-                    this.CurrentToken.ContextualKind != SyntaxKind.AsyncKeyword &&
-                    IsComplete(type))
-                {
-                    var misplacedModifier = this.CurrentToken;
-                    type = this.AddError(
-                        type,
-                        type.FullWidth + misplacedModifier.GetLeadingTriviaWidth(),
-                        misplacedModifier.Width,
-                        ErrorCode.ERR_BadModifierLocation,
-                        misplacedModifier.Text);
-
-                    return _syntaxFactory.IncompleteMember(attributes, modifiers.ToList(), type);
-                }
+                var afterTypeResetPoint = this.GetResetPoint();
 
-parse_member_name:;
-                // If we've seen the ref keyword, we know we must have an indexer, method, or property.
-                if (!sawRef)
+                try
                 {
```

Comment:
i dislike this entire method.  the complexity level is so high.  i would love if we could extract out someone useful here instead of just making this even harder to understand than it already is.

Reply:
I have removed this part in https://github.com/alrz/roslyn/commit/6a4d586859559d7a949aef8b64b518f4593f4e0b on top of https://github.com/dotnet/roslyn/pull/23533 long ago (probably needs update now)

Reply:
Agree :-) The method is very complex. A refactoring is desired.

---
In reply to: [252525562](https://github.com/dotnet/roslyn/pull/32983#discussion_r252525562) [](ancestors = 252525562)

Reply:
Thank you!

---
In reply to: [252779167](https://github.com/dotnet/roslyn/pull/32983#discussion_r252779167) [](ancestors = 252779167)